File: aws-s3/src/controller/UploadController.ts
import { Controller, Param, Post, UploadedFile, UseInterceptors } from '@nestjs/common';
import { FileInterceptor } from '@nestjs/platform-express';
import { ImageService } from '../service/ImageService';

@Controller()
export class UploadController {
  constructor(private imageService: ImageService) {}

  @Post('upload/:userId')
  @UseInterceptors(FileInterceptor('file'))
  async uploadFile(@Param('userId') userId: string, @UploadedFile() file: Express.Multer.File) {
    console.log(userId);
    await this.imageService.saveImage(userId, file);
  }
}

File: aws-s3/spring-boot/src/main/java/com/example/controller/UploadController.java
package com.example.controller;

import com.example.service.ImageService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RestController
@RequiredArgsConstructor
public class UploadController {

    private final ImageService imageService;

    @PostMapping("upload/{userId}")
    public void uploadFile(@PathVariable String userId, @RequestParam("file") MultipartFile file) {
        imageService.saveImage(userId, file);
    }
}
